lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -101,8 +109,10 @@ def all_schedules
 
       # clean the schedules as it exists in redis, useful for first setup?
       def clean_schedules(keys = non_persistent_schedules)
-        keys.each do |key|
-          remove_schedule(key)
+        redis.multi do
```

Comment:
Cleaning schedules must be done in a MULTI transaction every time.

This method is called two times in `resque-scheduler`: once during every instance's shutdown outside a transaction which causes schedules being loaded partially (e.g. if `reload_schedule!` is  called while an instance is shutting down, then we get a partial schedule) and the other time is inside a transaction when an instance is starting up